src/commands/deploy/deployTrialApp.ts

```diff
@@ -0,0 +1,52 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { ProgressLocation, window, workspace, WorkspaceFolder } from 'vscode';
+import { localGitDeploy } from 'vscode-azureappservice';
+import { IActionContext } from 'vscode-azureextensionui';
+import { WebAppTreeItem } from '../../../extension.bundle';
+import { SiteTreeItem } from '../../explorer/SiteTreeItem';
+import { TrialAppTreeItem } from '../../explorer/trialApp/TrialAppTreeItem';
+import { ext } from '../../extensionVariables';
+import { localize } from '../../localize';
+import { selectWorkspaceFolder } from '../../utils/workspace';
+import { showDeployCompletedMessage } from './showDeployCompletedMessage';
+
+export async function deployTrialApp(context: IActionContext, trialAppTreeItem: TrialAppTreeItem): Promise<void> {
+    const workspaceFolders: WorkspaceFolder[] | undefined = workspace.workspaceFolders;
+
+    let path: string;
+    let commit: boolean = false;
+
+    if (workspaceFolders?.length === 1) {
+        path = workspaceFolders[0].uri.fsPath;
+        commit = workspaceFolders[0].name === trialAppTreeItem.metadata.siteName;
+    } else {
+        path = await selectWorkspaceFolder('Select folder containing a repository to deploy');
+        const workspaceFolder: WorkspaceFolder | undefined = workspace.getWorkspaceFolder(vscode.Uri.file(path));
+        commit = workspaceFolder?.name === trialAppTreeItem.metadata.siteName;
+    }
+
+    const title: string = localize('deploying', 'Deploying to "{0}"... Check [output window](command:{1}) for status.', trialAppTreeItem.client.fullName, `${ext.prefix}.showOutputChannel`);
+    await window.withProgress({ location: ProgressLocation.Notification, title }, async () => {
+        await localGitDeploy(trialAppTreeItem.client, { fsPath: path, branch: 'RELEASE', commit: commit }, context);
+        return showDeployCompletedMessage(trialAppTreeItem);
+    });
+}
+
+export async function getDeployNodeWithTrialApp(context: IActionContext, target?: vscode.Uri | TrialAppTreeItem | SiteTreeItem): Promise<vscode.Uri | SiteTreeItem | TrialAppTreeItem | undefined> {
```

Comment:
Rather than having its own function, I think it makes sense to add this logic to the `getDeployNode`.  You should be able to change the showTreeItemPicker in there to allow you to select the trial app depending on the user's settings/logged in status.

Reply:
The reason I didn't add that logic to `getDeployNode` is that for trial apps we need to get the deploy node before we get the deploy fs path. 

And `getDeployNode` depends on the fs path. I think there might be a way to make this work but I believe that it would end up worse and less clear than it is now.

This problem spawns from the fact that normally we ask for the path first, but for trial apps we need to know the target first since they could be deploying to a trial app or a real app.